test/pngEncodeDecode_tests.cpp

```diff
@@ -0,0 +1,176 @@
+// C++ Standard Library
+#include <vector>
+#include <iostream>
+
+// C Standard Library
+#include <stdio.h>
+
+// UNIX stuff?
+#include <sys/stat.h>
+
+// Gtest
+#include <gtest/gtest.h>
+#include <gmock/gmock.h>
+
+// code to test
+#include <util/pngEncodeDecode.hpp>
+
+using ::testing::ElementsAreArray;
+
+namespace geometry {
+
+	/**
+	* @brief Overloads the << operator to print out the Cells in a vector of cells
+	* @param os The ostream object
+	* @param pixels The vector of cells to be printed
+	* @returns reference to the ostream object
+	*/	
+	std::ostream& operator<<(std::ostream& os, const std::vector<geometry::Cell>& pixels){
+		os << "\n";
+		for (const auto& pixel:pixels) {
+			os << "(" << pixel.x << "," << pixel.y << ")\n";
+		}
+		os << "\n";
+		return os;
+	}
+
+	/**
+	* @brief Overloads the == operator to allow equality check between two Cell objects
+	* @param lhs The cell on the left side of the == operator
+	* @param rhs The cell on the right side of the == operator
+	* @returns boolean on if the x and y values of the Cell are equal
+	*/	
+	bool operator==(const Cell &lhs,const Cell &rhs) {
+		return (lhs.x == rhs.x) && (lhs.y == rhs.y);
+	}
+
+	namespace utility {		
+
+		TEST(encode_test, file_existence){
+			// GIVEN a file name
+			const char* filename = "test.png";
+
+			// PREPARE by checking if the file exists, and if so deleting the file
+		    struct stat buffer;   
+		    if (stat ("test.png", &buffer) == 0) {system("rm ./test.png");}
```

Comment:
Let's use
https://en.cppreference.com/w/cpp/io/c/tmpfile
instead.
Do your best to adhere to RAII principles